### PR TITLE
run_ci.sh: git log what [base] version is being tested

### PR DESCRIPTION
**Pull request by marc-hb**
Zephyr Continuous Integration made the choice of rebasing pull requests
on its fast moving master branch before testing them. Unfortunately the
Shippable interface assumes the other choice[*] and highlights the SHA1
of the pull request _before_ rebasing which is very misleading.  For now
the only way to realize this disconnect is to scrutinize the build logs
and/or this source code (or tribal knowledge).

To mitigate and reduce a bit this confusion, run a short:
  git log -n 5 --oneline --decorate
before and after rebase.

Making it worse, Shippable sub-runs are not pegged to a common base
version. For instance the reason why shippable sub-run 35578.1 passed
while all other 35578.x sub runs failed is very likely because PR #13803
was merged to master in the mean time. This commit would have been able
to prove that.

[*] In an ideal world with infinite time and resources, Continuous
Integration would test pull requests BOTH alone and combined with the
moving tip.

### CODE REVIEW

**Comment by marc-hb** [comment]
To see how what difference it makes in the builds logs click on:

  Shippable -> Details -> Success -> build_ci -> 'if [ "$IS_PULL_REQUEST" = "true" ] ... '

... and then scroll up a lot.

Not exactly showing what version CI is testing front and center but much better than not showing it at all.

File: scripts/ci/run_ci.sh [hunk 1]
```diff
@@ -96,9 +96,18 @@ if [ -n "$MAIN_CI" ]; then
 	fi
 	source zephyr-env.sh
 	SANITYCHECK="${ZEPHYR_BASE}/scripts/sanitycheck"
+
+	# Possibly the only record of what exact version is being tested:
+	short_git_log='git log -n 5 --oneline --decorate --abbrev=12 '
+
 	if [ -n "$PULL_REQUEST_NR" ]; then
+		$short_git_log $REMOTE/${BRANCH}
```
Comment (ulfalizer) [inline scripts/ci/run_ci.sh]: Nit: Could make it `$REMOTE/$BRANCH` to have a consistent style.

File: scripts/ci/run_ci.sh [hunk 2]
```diff
@@ -96,9 +96,18 @@ if [ -n "$MAIN_CI" ]; then
...
+		# Now let's pray this script is being run from a
+		# different location
+# https://stackoverflow.com/questions/3398258/edit-shell-script-while-its-running
```
Comment (ulfalizer) [inline scripts/ci/run_ci.sh]: Was this unindented to work around some CI line length check? Otherwise, I'd just let the line be as long as it needs.

**Review by ulfalizer — APPROVED** [review]
Don't know that much about these scripts, but seems fine to me.

File: scripts/ci/run_ci.sh (hunk 2 above)
Comment (andyross) [reply scripts/ci/run_ci.sh]: No opinion as to alignment or whether or not the joke belongs in the source, but can I say that's the most hilariously wrong accepted answer I've ever seen on SO.

**Review by andyross — APPROVED** [review]

File: scripts/ci/run_ci.sh (hunk 2 above)
Comment (marc-hb) [reply scripts/ci/run_ci.sh]: > Was this unindented to work around some CI line length check?

It was partly trying to dodge some gitlint or other bullet (which... doesn't actually exist, just checked) and partly not to write too far right: this sort of check exists for a reason! I don't really mind either way and will indent.

@andyross : I was also very amused by that SO answer, however I'm sorry it and/or my phrasing seemed to eclipse the "value" I tried to add with this comment: I once lost hair before finally understanding that the (run-time copy of the) shell script I was trying to debug was being modified while it was running. A comment like this one could have saved some of that hair.
Any rephrasing suggestion appreciated.

File: scripts/ci/run_ci.sh (hunk 1 above)
Comment (marc-hb) [reply scripts/ci/run_ci.sh]: This current version is consistent with... the inconsistent code 4 lines below :-( 
I will change just this line (I'd rather not fix the other line and take its git blame)

**Review by marc-hb — COMMENTED** [review]
github requests something here

**Pull request merged**